tests/resources/promise-rejection-events.js

```diff
@@ -446,14 +446,14 @@ async_test(function(t) {
    'all inside a setTimeout');
 
 //
-// Positive unhandledrejection/rejectionhandled tests with delayed attachment
+// Negative unhandledrejection/rejectionhandled tests with task-delayed attachment
 //
 
 async_test(function(t) {
   var e = new Error();
   var p;
 
-  onUnhandledSucceed(t, e, function() { return p; });
+  onUnhandledFail(t, function() { return p; });
```

Comment:
Oh, hmm, the spec definitely makes a copy as-is. But that seems counter to what we agreed, and what the OP of this pull request says ("option 3"). Argh, trying to reload our context...

Does it work to just use the same global list throughout all algorithms, and clear it after the task loop? Or does that have possible reentrancy issues? I guess we could do what was suggested in option 3 directly, and re-check each promise's internal state to see if it's been handled.